mod-source-record-storage-server/src/main/java/org/folio/consumers/DataImportKafkaHandler.java

```diff
@@ -76,34 +105,63 @@ public Future<String> handle(KafkaConsumerRecord<String, byte[]> targetRecord) {
-          .map(profileSnapshot -> EventManager.handleEvent(eventPayload, profileSnapshot))
-          .orElse(CompletableFuture.failedFuture(new EventProcessingException(format("Job profile snapshot with id '%s' does not exist", jobProfileSnapshotId)))))
+        .thenCompose(snapshotOptional -> {
+          printLogInfo(DEBUG, format("handle:: Is snapshot profile with id: '%s' is received: '%s'", jobProfileSnapshotId, snapshotOptional.isPresent()), debugInfo);
+          return snapshotOptional.map(profileSnapshot -> {
+              printLogInfo(DEBUG, "handle:: Handle event by EventManager", debugInfo);
+              return EventManager.handleEvent(eventPayload, profileSnapshot);
+            })
+            .orElse(CompletableFuture.failedFuture(new EventProcessingException(format("Job profile snapshot with id '%s' does not exist", jobProfileSnapshotId))));
+        })
         .whenComplete((processedPayload, throwable) -> {
           if (throwable != null) {
+            printLogInfo(ERROR, format("handle:: Failed to process data import event payload from topic %s", targetRecord.topic()), throwable, debugInfo);
             promise.fail(throwable);
           } else if (DI_ERROR.value().equals(processedPayload.getEventType())) {
-            promise.fail(format("handle:: Failed to process data import event payload from topic '%s' by jobExecutionId: '%s' with recordId: '%s' and chunkId: '%s' ", targetRecord.topic(),
-              eventPayload.getJobExecutionId(), recordId, chunkId));
+            printLogInfo(ERROR, format("handle:: Failed to process data import event payload from topic %s", targetRecord.topic()), debugInfo);
+            promise.fail(format("Failed to process data import event payload from topic '%s' by jobExecutionId: '%s' with recordId: '%s' and chunkId: '%s' ",  targetRecord.topic(), jobExecutionId, recordId, chunkId));
           } else {
+            printLogInfo(INFO, format("handle:: Successfully processed data import event payload from topic %s", targetRecord.topic()), debugInfo);
             promise.complete(targetRecord.key());
           }
         });
       return promise.future();
     } catch (Exception e) {
-      LOGGER.warn("handle:: Failed to process data import kafka record from topic '{}' with recordId: '{}' and chunkId: '{}' ", targetRecord.topic(), recordId, chunkId, e);
+      LOGGER.error("handle:: Failed to process data import kafka record from topic '{}' with recordId: '{}' and chunkId: '{}' ", targetRecord.topic(), recordId, chunkId, e);
       return Future.failedFuture(e);
     }
   }
+
+  private static void printLogInfo(Level level, String message, String... args) {
+    printLogInfo(level, message, null, args);
+  }
+
+  private static void printLogInfo(Level level, String message, Throwable tr, String... args) {
+    String formattedMessage;
+    if (tr != null) {
+      formattedMessage = format(ERROR_LOG_MESSAGE_TEMPLATE, message, args[0], args[1], args[2], args[3], tr.getMessage());
```

Comment:
propose to pass the whole exception to log instead of the message so that we could also have a stack trace in case an investigation is needed

Reply:
Completed. Thanks!